### ISSUE TITLE
Add meta-strategy selector to route signals based on market conditions

### ISSUE THREAD

**Issue by braxtondiggs** [braxtondiggs/Chansey#299]
## Summary

Build a meta-strategy selection layer that predicts which of the existing 11 trading strategies will perform best given current market conditions, and routes execution accordingly.

## Background

Different strategies excel in different market conditions — Confluence crushes trending markets but bleeds in choppy ones, Triple EMA does the opposite. Currently, strategy selection is manual or static. A meta-selector that learns "given these conditions, strategy X has historically outperformed" would act as an intelligent routing layer on top of existing strategies.

This is a much easier ML problem than price prediction — we're not predicting the market, we're predicting which of our own tools works best right now, using data we already collect.

## Requirements

- [ ] Feature engineering from existing data: volatility (ATR), trend strength (ADX), volume profile, correlation between coins, market regime
- [ ] Classifier trained on historical backtest performance data per strategy per market condition
- [ ] Scoring/ranking output: "Given current conditions, rank strategies by expected performance"
- [ ] Integration with pipeline orchestration to influence strategy deployment decisions
- [ ] Continuous learning — retrain as new backtest/live performance data accumulates

### Training Data (already exists)

- `AlgorithmPerformance` — per-algorithm performance metrics
- `BacktestPerformanceSnapshot` — periodic portfolio state during backtests
- `MarketRegime` — market condition classification (bull/bear/sideways)
- `StrategyScore` — score tracking per strategy
- `OptimizationResult` — parameter optimization outcomes
- `WalkForwardWindow` — walk-forward analysis results

## Technical Notes

- **Can start in TypeScript** — a weighted scoring model or simple decision tree can be implemented without Python
- Graduate to Python (gradient boosted trees via XGBoost/LightGBM) if the simple version shows promise
- Feature vector: `[volatility, trend_strength, volume_zscore, regime, correlation_avg, ...]`
- Target variable: strategy ranking by Sharpe ratio over next N-day window
- Evaluation: walk-forward cross-validation using existing backtest infrastructure

### MVP Approach (TypeScript, no Python)

```typescript
// Simple weighted scoring based on regime + volatility
function selectStrategy(regime: MarketRegime, volatility: number): Algorithm {
  const scores = strategies.map(s => ({
    strategy: s,
    score: historicalPerformance[s.id][regime] * regimeWeight
         + volatilityFit[s.id](volatility) * volWeight
  }));
  return scores.sort((a, b) => b.score - a.score)[0].strategy;
}
```

### Advanced Approach (Python ML)

```python
# Gradient boosted trees predicting strategy performance
model = XGBClassifier()
model.fit(market_features, best_strategy_labels)
predicted_best = model.predict(current_features)
```

## Acceptance Criteria

- [ ] MVP: TypeScript weighted scoring model using historical performance by regime
- [ ] Backtestable: Can evaluate "would meta-selection have improved overall returns?"
- [ ] Non-breaking: Existing manual strategy selection still works, meta-selector is opt-in
- [ ] Dashboard visibility: Show why a strategy was selected (feature importance / reasoning)
- [ ] Measurable improvement over static strategy selection in backtests

## Additional Context

- 11 existing strategies: RSI, MACD, EMA, Bollinger Bands, ATR-Trailing, Confluence, RSI-MACD combo, Triple EMA, Bollinger Squeeze, RSI Divergence, etc.
- Pipeline flow: `OPTIMIZE → HISTORICAL → LIVE_REPLAY → PAPER_TRADING → COMPLETED` — meta-selector could influence which strategies enter the pipeline
- Market regime module: `apps/api/src/market-regime/`
- Algorithm performance: `apps/api/src/algorithm/`
- Strategy evaluation task: `apps/api/src/tasks/strategy-evaluation.task.ts`
